homeassistant/components/picnic/sensor.py

```diff
@@ -0,0 +1,123 @@
+"""Definition of Picnic sensors."""
+
+from typing import Any, Optional
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_ATTRIBUTION
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.typing import StateType
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import ADDRESS, ATTRIBUTION, CONF_COORDINATOR, DOMAIN, SENSOR_TYPES
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, config_entry: ConfigEntry, async_add_entities
+):
+    """Set up Picnic sensor entries."""
+    picnic_coordinator = hass.data[DOMAIN][config_entry.entry_id][CONF_COORDINATOR]
+
+    # Fetch initial data so we have data when entities subscribe
+    await picnic_coordinator.async_refresh()
```

Comment:
We should do this in `picnic/__init__.py`. That way we can raise `ConfigEntryNotReady` if the fetch fails (or even `return False` if auth is no longer valid).